apis/config/v1beta1/configuration_types.go

```diff
@@ -319,6 +319,10 @@ type Integrations struct {
 	//  - "kubeflow.org/xgboostjob"
 	//  - "pod"
 	Frameworks []string `json:"frameworks,omitempty"`
+	// IntegrationRetryInterval refers to the duration to wait before retrying the API request.
+	// Defaults to 20 seconds if not set.
+	// +optional
+	IntegrationRetryInterval *metav1.Duration `json:"integrationRetryInterval,omitempty"`
```

Comment:
Actually, I'm not sure we want to expose this configuration at the surface of the API. @ChristianZaccaria do you have a use case for changing the value, or we could come up with a satisfying default which we set as a const? I would prefer not to expose it if we don't have a clear need, and we do research about other approaches.

The reason is - I'm not sure this is the optimal approach (see another comment about a CRDs watch), and exposing this at the API level would sort of commit us to it (deprecation of API is hard).

Reply:
I agree, please see the other comment for details. Should I remove the latest commit and remove this as a config option, and just have minimal changes to include the default value?

Reply:
Please remove it from API (which is largely the last commit). I'm ok to keep it as an iinput option to some function that is used for test purposes to make the tests run fast.